cmd/initContainer/main.go

```diff
@@ -119,8 +121,63 @@ func main() {
 	failure := false
 
 	run := func() {
-		_, err := kubeClientLeaderElection.CoordinationV1().Leases(getKyvernoNameSpace()).Get(ctx, "kyvernopre-lock", v1.GetOptions{})
+		certProps, err := tls.GetTLSCertProps(clientConfig)
+		if err == nil {
+			depl, err := client.GetResource("", "Deployment", getKyvernoNameSpace(), config.KyvernoDeploymentName)
+
+			deplHash := ""
+			if err == nil {
+				deplHash = fmt.Sprintf("%v", depl.GetUID())
+			} else {
+				log.Log.Info("failed to fetch deployment '%v': %v", config.KyvernoDeploymentName, err.Error())
+				os.Exit(1)
+			}
+
+			name := tls.GenerateRootCASecretName(certProps)
+			secretUnstr, err := client.GetResource("", "Secret", getKyvernoNameSpace(), name)
+			if err == nil {
+				if tls.CanAddAnnotationToSecret(deplHash, secretUnstr) {
+					secretUnstr.SetAnnotations(map[string]string{tls.MasterDeploymentUID: deplHash})
+					_, err = client.UpdateResource("", "Secret", certProps.Namespace, secretUnstr, false)
+
+					if err != nil {
+						log.Log.Info("failed to update cert: %v", err.Error())
+						os.Exit(1)
+					}
+				}
+			} else {
+				log.Log.Info("failed to fetch secret '%v': %v", name, err.Error())
+
+				if !errors.IsNotFound(err) {
+					os.Exit(1)
+				}
+			}
+
+			name = tls.GenerateTLSPairSecretName(certProps)
+			secretUnstr, err = client.GetResource("", "Secret", getKyvernoNameSpace(), name)
+			if err == nil {
+				if tls.CanAddAnnotationToSecret(deplHash, secretUnstr) {
+					secretUnstr.SetAnnotations(map[string]string{tls.MasterDeploymentUID: deplHash})
+					_, err = client.UpdateResource("", "Secret", certProps.Namespace, secretUnstr, false)
+
+					if err != nil {
+						log.Log.Info("failed to update cert: %v", err.Error())
+						os.Exit(1)
+					}
+				}
+			} else {
+				log.Log.Info("failed to fetch secret '%v': %v", name, err.Error())
+
+				if !errors.IsNotFound(err) {
+					os.Exit(1)
+				}
+			}
+		} else {
+			log.Log.Info("failed to get cert properties: %v", err.Error())
+			os.Exit(1)
```

Comment:
What about this error? Is it appropriate to exit for all different errors? What if the apiServer is temporarily unavailable?